x-pack/plugins/security_solution/public/common/components/ml_popover/hooks/use_enable_data_feed.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useReducer } from 'react';
+import { useAppToasts } from '../../../hooks/use_app_toasts';
+import { METRIC_TYPE, TELEMETRY_EVENT, track } from '../../../lib/telemetry';
+import { setupMlJob, startDatafeeds, stopDatafeeds } from '../api';
+import type { SecurityJob } from '../types';
+import * as i18n from './translations';
+
+function mlPopoverReducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'loading': {
+      return {
+        ...state,
+        isLoading: true,
+      };
+    }
+    case 'success': {
+      return {
+        ...state,
```

Comment:
If the State is only ever { isLoading: boolean } why for each case we spread the state? wouldnt it be just the one value, that we end up updating anyway? Also why use a reducer if the state is so simple? Not that it's wrong, just curious

Reply:
Good catch. You are right. We don't have to spread the state. The state used to be more complex, but I deleted some properties in this PR.

I update the code to use `useState`.